testing/mongo-c-driver/APKBUILD

```diff
@@ -0,0 +1,61 @@
+# Maintainer: Leonardo Arena <rnalrd@alpinelinux.org>
+pkgname=mongo-c-driver
+pkgver=1.14.0
+pkgrel=0
+pkgdesc="Client library written in C for MongoDB"
+url="https://github.com/mongodb/mongo-c-driver"
+arch="all"
+license="Apache-2.0"
+makedepends="openssl-dev cyrus-sasl-dev icu-dev py3-sphinx cmake"
+subpackages="${pkgname}-static ${pkgname}-dev ${pkgname}-doc libbson"
+source=${pkgname}-${pkgver}.tar.gz::https://github.com/mongodb/${pkgname}/archive/${pkgver}.tar.gz
+builddir="${srcdir}/${pkgname}-${pkgver}"
+
+build() {
+	cmake \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_INSTALL_PREFIX:PATH=/usr \
+		-DCMAKE_INSTALL_LIBDIR:PATH=lib \
+		-DENABLE_BSON:STRING=ON \
+		-DENABLE_MONGOC:BOOL=ON \
+		-DENABLE_SSL:STRING=OPENSSL \
+		-DENABLE_AUTOMATIC_INIT_AND_CLEANUP:BOOL=OFF \
+		-DENABLE_MAN_PAGES:BOOL=ON \
+		-DENABLE_TESTS:BOOL=ON \
+		-DENABLE_EXAMPLES:BOOL=OFF \
+		-DSPHINX_EXECUTABLE:STRING=/usr/bin/sphinx-build-3 \
+		-DCMAKE_SKIP_RPATH:BOOL=ON \
+		-DENABLE_SHM_COUNTERS:BOOL=ON \
+		-DENABLE_SASL:STRING=CYRUS \
+		-DENABLE_ICU:STRING=ON \
+		-DENABLE_CRYPTO_SYSTEM_PROFILE:BOOL=ON \
+		-DENABLE_UNINSTALL:BOOL=OFF \
+		-DBUILD_VERSION:STRING=${pkgver} \
+		.
+	make
+}
+
+check() {
+	export MONGOC_TEST_SKIP_MOCK=on
+	export MONGOC_TEST_SKIP_SLOW=on
+	export MONGOC_TEST_SKIP_LIVE=on
+	make check
+}
+
+package() {
+	make DESTDIR="${pkgdir}" install
+}
+
+static() {
+	pkgdesc="Static libs for mongo-c-driver and libbson"
+	mkdir -p "${subpkgdir}/usr/lib"
+	find "${pkgdir}/usr/lib/" -name "*.a" -type f -exec mv "{}" "${subpkgdir}/usr/lib" \;
```

Comment:
Why not use `default_static` here instead?

Reply:
Updated.
Before I stuck with improper static files packaging. If inside `subpackages` variable `-static` package defined after `-dev` package and used default packaging, then `abuild` script fail.

```diff
diff --git a/testing/mongo-c-driver/APKBUILD b/testing/mongo-c-driver/APKBUILD
index 0cf973fd8f..83f7fcde08 100644
--- a/testing/mongo-c-driver/APKBUILD
+++ b/testing/mongo-c-driver/APKBUILD
@@ -7,7 +7,7 @@ url="https://github.com/mongodb/mongo-c-driver"
 arch="all"
 license="Apache-2.0"
 makedepends="openssl-dev cyrus-sasl-dev icu-dev py3-sphinx cmake"
-subpackages="${pkgname}-static ${pkgname}-dev ${pkgname}-doc libbson"
+subpackages="${pkgname}-dev ${pkgname}-static ${pkgname}-doc libbson"
 source=${pkgname}-${pkgver}.tar.gz::https://github.com/mongodb/${pkgname}/archive/${pkgver}.tar.gz
 builddir="${srcdir}/${pkgname}-${pkgver}"
```

Where I can fill improvement request.

Reply:
Are you asking about the ordering? If so, see: https://github.com/alpinelinux/abuild/pull/72

Reply:
yes, about this.
I will mark this conversation as resolved.